docs/source/index.rst

```diff
@@ -13,11 +13,12 @@ OpenMC was originally developed by members of the `Computational Reactor Physics
 Group`_ at the `Massachusetts Institute of Technology`_ starting
 in 2011. Various universities, laboratories, and other organizations now
 contribute to the development of OpenMC. For more information on OpenMC, feel
-free to send a message to the User's Group `mailing list`_.
+free to send a message to the User's Group `mailing list`_. Documentation for the latest version of the develop branch can be found on `Read the Docs`_.
```

Comment:
I would say "latest developmental version" and wrap this line to stay consistent.